src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -202,19 +275,350 @@ protected override ODataRequestMessageWrapper CreateRequestMessage(string method
+                            }
+                        }
+                        else 
+                        {
+                            descriptor = relatedDescriptors[0];
+                            this.adjacentDescriptors.Push((0, relatedDescriptors));
+                        }
+                    }          
+                }
+                 
+                if (descriptor != null) 
+                {
+                    this.SaveResultProcessed(descriptor, HasFailedOperation(materializerState));
+
+                    operationResponse = new ChangeOperationResponse(this.cachedResponse.Headers, descriptor)
+                    {
+                        StatusCode = (int)this.cachedResponse.StatusCode
+                    };
+
+                    this.materializerStateForDescriptor.Add(descriptor, materializerState);
+
+                    if (this.cachedResponse.Exception != null)
+                    {
+                        operationResponse.Error = this.cachedResponse.Exception;
+
+                        if (ex == null)
+                        {
+                            ex = this.cachedResponse.Exception;
+                        }
+                    }
+                    else
+                    {
+#if DEBUG
+                        this.HandleOperationResponse(descriptor, this.cachedResponse.Headers, this.cachedResponse.StatusCode);
+#else
+                        this.HandleOperationResponse(descriptor, this.cachedResponse.Headers);
+#endif
+                    }
+
+                    if (isTopLevelResponse)
+                    {
+                        this.operationResponses.Add(operationResponse);
+                        this.operationResponseTracker.Push(this.operationResponses);
+                    }
+                    else
+                    {
+                        List<OperationResponse> lastOperationResponses = this.operationResponseTracker.Peek();
+                        lastOperationResponses.Last().NestedResponses.Add(operationResponse);
+
+                        if (this.bulkUpdateGraph.GetRelatedDescriptors(descriptor).Count == 0)
+                        {
+                            this.operationResponseTracker.Pop();
+                            this.operationResponseTracker.Push(lastOperationResponses);
+                        }
+                        else
+                        {
+                            this.operationResponseTracker.Pop();
+                            this.operationResponseTracker.Push(lastOperationResponses.Last().NestedResponses);
+                        }
+                    }
+                }  
+            }
+            catch (InvalidOperationException e)
+            {
+                ex = e;
+            }
+
+            if (ex != null)
+            {
+                throw new DataServiceRequestException(Strings.DataServiceException_GeneralError, ex, dataServiceResponse);
+            }

```

Comment:
Unless there's something I'm missing, we can do it this way:
```suggestion
            catch (InvalidOperationException e)
            {
                throw new DataServiceRequestException(Strings.DataServiceException_GeneralError, e, dataServiceResponse);
            }
```

Reply:
We only throw this exception is `ex` is `null`

Reply:
@ElizabethOkerio `ex` is initialized from `e` and it won't be null in the `catch` block.

Reply:
I think there is some other place where `ex` is initialized not just from `e`